### DISCUSSION TITLE
Which's the correct way to get the voyager data from the DB?

### DISCUSSION THREAD

**Discussion by lamualfa** [thedevdojo/voyager#5531] [Q&A] [answered]
### 1. Getting manually using Model (_Laravel Eloquent_)

**Example**

```php
News::select('title', 'description')->where('id', 'xxx')->get()->toArray();
```

**Pros**

- More control
- More flexible

**Cons**

- Does not implement user roles
- Does not group the data likes [Menu & Menu Items](https://voyager-docs.devdojo.com/core-concepts/menus-and-menu-builder)
- Does not parse Input File data from string to object (array)

### 2. Getting manually using `DB::table`

For Voyager built-in table, sometimes I use this method to fetch the data from them.

**Example**

```php
DB::table('menu_items')->where('menu', menu_id)->get()->toArray();
```

again, the problem with this method is also the same as the first method.

### 3 Or else

**Comment by ousid** [comment, marked as answer]
Hey, @lamualfa 

I don't get voyager data__, you mean by that, you need to get the records from the database?

Alright, there is a `Voyager::model('the model')` if you are looking to fetch the qualified name model, or if you want to set some rules, there are  a bunch of methods you can use.

You may take a look here:
https://github.com/the-control-group/voyager/blob/1.5/src/Widgets/UserDimmer.php

Or here:
https://voyager-docs.devdojo.com/core-concepts/roles-and-permissions#customize-controller
